### PR TITLE
[TASK] Add workspace persistence in-depth scenarios

### PR DESCRIPTION
**Pull request by ohader**

### CODE REVIEW

File: Documentation/ApiOverview/Workspaces/Index.rst [hunk 1]
```diff
@@ -459,3 +459,252 @@ record is selected it should simply be discarded in case shown in
...
+
+The following section represents how database records are actually persisted in a database
+table for different scenarios and previously performed actions.
+
+Placeholders
+------------
+
+Workspace placeholders are stored in field :code:`t3ver_state` which can have the following values:
+
+* `-1`: new placeholder version - the workspace pendant for a new placeholder (see value `1`)
```
Comment (DanielSiepmann) [inline Documentation/ApiOverview/Workspaces/Index.rst]: You could use a definition list. That might be a more reader friendly representation.

Comment (ohader) [reply Documentation/ApiOverview/Workspaces/Index.rst]: Do you have an example how to do that?

Comment (DanielSiepmann) [reply Documentation/ApiOverview/Workspaces/Index.rst]: Forgot that, already had tab open: https://docs.typo3.org/m/typo3/book-extbasefluid/master/en-us/b-ExtbaseReference/Index.html#localization

This contains multiple examples for APIs, e.g. annotations with a but of code, or APIs of query builder with less code.

Comment (ohader) [reply Documentation/ApiOverview/Workspaces/Index.rst]: Will add these changes as additional commit soon

**Comment by ohader** [comment]
Since a couple of changes are planned for v10, it probably would be good to back-port this change to v9 as well. Shall I create a new pull-request for that?

**Comment by DanielSiepmann** [comment]
Yes, thanks. Your PR would be welcome :) Would save me the work.

**Comment by ohader** [comment]
Follow-up in #582

**Comment by ohader** [comment]
> Yes, thanks. Your PR would be welcome :) Would save me the work.

Sure, I'll create a combined back-port to v9 of this PR #579 and new #582 once merged to master

**Pull request merged**